receiver/awscontainerinsightreceiver/internal/stores/podstore.go

```diff
@@ -90,27 +91,28 @@ func newMapWithExpiry(ttl time.Duration) *mapWithExpiry {
 	}
 }
 
-type replicaSetInfo interface {
-	ReplicaSetToDeployment() map[string]string
+type replicaSetInfoProvider interface {
+	GetReplicaSetClient() k8sclient.ReplicaSetClient
 }
 
 type podClient interface {
 	ListPods() ([]corev1.Pod, error)
 }
 
 type PodStore struct {
+	ctx              context.Context
```

Comment:
Storing a `context.Context` in a struct is generally discouraged.  Can you do this another way?  If not, document here why this is necessary?

Reply:
I moved the context to `store.go` and still save it in the struct (because I need to use the context when calling the Decorate function). I added some explanation there as well.